Migrate legacy test TermsAndConditionsTest and refactor related class

matks opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Migrate and refactor TermsAndConditionsTest, refactor class TermsAndConditions
| Type?         | refacto
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  |Travis should be green

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/12249)
<!-- Reviewable:end -->

matks committed:
Migrate legacy test TermsAndConditionsTest and refactor class TermsAndConditions

Quetzacoalt91 commented:
Is this class used somewhere in the project @matks? The QA team could check no feature is impacted by this change.

matks commented:
It's used in Checkout process (see class `ConditionsToApproveFinderCore`) to display Terms and Conditions to the customer and get him to approve them using a checkbox.

In theory I have only added phpdoc, renamed variables and moved some code without modifying it. Only https://github.com/PrestaShop/PrestaShop/pull/12249/files#diff-5b1a7682c21540cdf6841600e9cad867R106 is modified significantly because I used `sprintf`.

Since the unit tests still pass I think I broke nothing ... but who knows 😄 ?

mickaelandrieu commented:
Thanks @matks
